providers/gitlab_test.go

```diff
@@ -62,8 +63,9 @@ func TestGitLabProviderBadToken(t *testing.T) {
 	bURL, _ := url.Parse(b.URL)
 	p := testGitLabProvider(bURL.Host)
 
+	ctx := context.TODO()
```

Comment:
Please do the others in this file too
```suggestion
	ctx := context.Background()
```